src/components/component.ts

```diff
@@ -479,29 +480,51 @@ module Plottable {
      * @returns The calling Component.
      */
     public detach() {
+      this.parent(null);
+
       if (this._isAnchored) {
         this._element.remove();
       }
+      this._isAnchored = false;
 
-      var parent: ComponentContainer = this._parent();
+      this._onDetachCallbacks.callCallbacks(this);
+      return this;
+    }
 
-      if (parent != null) {
-        parent.remove(this);
-      }
-      this._isAnchored = false;
-      this._parentElement = null;
+    /**
+     * Adds a callback to be called when th Component is detach()-ed.
+     *
+     * @param {ComponentCallback} callback The callback to be added.
+     * @return {Component} The calling Component.
+     */
+    public onDetach(callback: ComponentCallback) {
```

Comment:
That doesn't see quite right. I'm also not sure that should be the case for the `anchor()` callback. Instead we should have an `anchored()` method or `inDOM()` method.